### PR TITLE
Fix bugs and add the conditions to the preset config file

### PR DESCRIPTION
**Pull request by fansaien**
- Can init from the preset config file now
- The preset file supports conditions
- merged from https://github.com/rainlab/notify-plugin/pull/16/files Use Laravel's Request object to change request -POST- values #16

### CODE REVIEW

**Comment by w20k** [comment]
Hello, @fansaien good to see you back! Will do a quick peer review, tomorrow 👍 Can't say about testing, though)

File: formwidgets/ActionBuilder.php [hunk 1]
```diff
@@ -245,17 +246,21 @@ public function setCacheActionData($action)
 
         $cache[$action->id] = json_encode($this->makeCacheActionData($action));
 
-        $_POST['action_data'] = $cache;
+        $requestData = Request::all();
+        array_set($requestData, 'action_data' , $cache);
+        Request::merge($requestData);
```
Comment (bennothommo) [inline formwidgets/ActionBuilder.php]: @fansaien As per [this comment](https://github.com/rainlab/notify-plugin/pull/16#issuecomment-495843838), could you try rewriting the above as:

```php
Request::merge([
    'action_data' => $cache
]);
```
and do the same for the code in the `restoreCacheActionDataPayload` method and `ConditionBuilder`.

File: updates/version.yaml [hunk 2]
```diff
@@ -5,3 +5,4 @@
     - create_rule_conditions_table.php
     - create_rule_actions_table.php
 1.0.2: Fixes crashing bug.
+1.0.3: Fixes POST bug and supports the conditions settings in the preset config file
```
Comment (bennothommo) [inline updates/version.yaml]: @fansaien Could you remove this change from the current PR? We'll bump up the version number and modify `version.yaml` when ready. :)

File: formwidgets/ActionBuilder.php (hunk 1 above)
Comment (fansaien) [reply formwidgets/ActionBuilder.php]: Please review the new version, thanks

**Pull request merged**